doc/dox_comments/header_files/aes.h

```diff
@@ -1974,3 +1974,1696 @@ int wc_AesCtsDecryptUpdate(Aes* aes, byte* out, word32* outSz,
+    if (ret != 0) {
+        // failed to initialize
+    }
+    wc_AesFree(&aes);
+    \endcode
+
+    \sa wc_AesGcmInit
+    \sa wc_AesGcmEncryptUpdate
+*/
+int wc_AesGcmEncryptInit(Aes* aes, const byte* key, word32 len,
+                         const byte* iv, word32 ivSz);
+
+/*!
+    \ingroup AES
+    \brief This function initializes an AES GCM cipher for encryption and
+    outputs the IV. This is useful when part of the IV is generated
+    internally. Must call wc_AesGcmSetIV() before this function to set
+    the fixed part of the IV.
+
+    \return 0 On success.
+    \return BAD_FUNC_ARG If aes, ivOut is NULL, or if ivOutSz doesn't
+    match the cached nonce size.
+
+    \param aes pointer to the AES structure to initialize
+    \param key pointer to the key buffer, or NULL to skip key setting
+    \param len length of the key in bytes
+    \param ivOut pointer to buffer to receive the complete IV
+    \param ivOutSz length of the IV output buffer in bytes
+
+    _Example_
+    \code
+    Aes aes;
+    byte key[16] = { }; // 128-bit key
+    byte ivFixed[4] = { }; // fixed part of IV
+    byte ivOut[12];
+    WC_RNG rng;
+
+    wc_InitRng(&rng);
+    wc_AesInit(&aes, NULL, INVALID_DEVID);
+    wc_AesGcmSetIV(&aes, 12, ivFixed, 4, &rng);
+    int ret = wc_AesGcmEncryptInit_ex(&aes, key, 16, ivOut, 12);
+    if (ret != 0) {
+        // failed to initialize
+    }
+    wc_AesFree(&aes);
+    wc_FreeRng(&rng);
+    \endcode
+
+    \sa wc_AesGcmSetIV
+    \sa wc_AesGcmEncryptUpdate
+*/
+int wc_AesGcmEncryptInit_ex(Aes* aes, const byte* key, word32 len,
+                            byte* ivOut, word32 ivOutSz);
+
+/*!
+    \ingroup AES
+    \brief This function performs an update step of AES GCM encryption.
+    It processes plaintext and/or additional authentication data (AAD)
+    in a streaming fashion.
+
+    \return 0 On success.
+    \return BAD_FUNC_ARG If aes is NULL, or if parameters are invalid.
+
+    \param aes pointer to the AES structure
+    \param out pointer to buffer to store ciphertext (can be NULL if sz=0)
+    \param in pointer to plaintext to encrypt (can be NULL if sz=0)
+    \param sz length of plaintext in bytes
+    \param authIn pointer to additional authentication data (can be NULL)
+    \param authInSz length of AAD in bytes
+
+    _Example_
+    \code
+    Aes aes;
+    byte key[16] = { }; // 128-bit key
+    byte iv[12] = { }; // nonce
+    byte plaintext[100] = { }; // data
+    byte ciphertext[100];
+    byte aad[20] = { }; // additional data
+
+    wc_AesInit(&aes, NULL, INVALID_DEVID);
+    wc_AesGcmEncryptInit(&aes, key, 16, iv, 12);
+    int ret = wc_AesGcmEncryptUpdate(&aes, ciphertext, plaintext, 100,
+                                     aad, 20);
+    if (ret != 0) {
+        // encryption failed
+    }
+    wc_AesFree(&aes);
+    \endcode
+
+    \sa wc_AesGcmEncryptInit
+    \sa wc_AesGcmEncryptFinal
+*/
+int wc_AesGcmEncryptUpdate(Aes* aes, byte* out, const byte* in, word32 sz,
```

Comment:
The wc_AesGcmEncryptUpdate and DecryptUpdate are already commented in `wolfcrypt/src/aes.c`.

They have these additional comments that are important:

```
 * All the AAD must be passed to update before the plaintext.                   
 * Last part of AAD can be passed with first part of plaintext.                 
 *                                                                              
 * Must set key and IV before calling this function.                            
 * Must call wc_AesGcmInit() before calling this function.
 * ```